### PR TITLE
ignore init sql

### PR DESCRIPTION
**Pull request by ZhangMan1126**
ignore initialization sql in sql heuristics

### CODE REVIEW

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/EMController.java [hunk 1]
```diff
@@ -507,6 +509,8 @@ public Response executeDatabaseCommand(DatabaseCommandDto dto, @Context HttpServ
 
         try {
 
+            ExecutionTracer.executingInitSql = true;
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/EMController.java:512]: unfortunately, I just realized this approach does not work with External Drivers :(  
we need to discuss this

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/EMController.java (hunk 1 above)
Comment (ZhangMan1126) [reply client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/EMController.java:512]: ok.

**Comment by arcuri82** [comment]
thanks. unfortunately we do not have (yet) proper E2E scaffolding for external drivers :(  just make sure it seems working with some manual testing using actual external drivers (eg from EMB)

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/ExternalSutController.java [hunk 2]
```diff
@@ -364,6 +364,13 @@ public void setKillSwitch(boolean b) {
         ExecutionTracer.setKillSwitch(b);// store info locally as well, to avoid needing to do call to fetch current value
     }
 
+    @Override
+    public void setExecutingInitSql(boolean executingInitSql) {
+        checkInstrumentation();
+        serverController.setExecutingInitSql(executingInitSql);
+        // Man: why there are two set, need to check
+        ExecutionTracer.setKillSwitch(executingInitSql);
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/ExternalSutController.java:372]: :)

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/EMController.java [hunk 3]
```diff
@@ -507,6 +508,8 @@ public Response executeDatabaseCommand(DatabaseCommandDto dto, @Context HttpServ
 
         try {
 
+            sutController.setExecutingInitSql(true);
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/EMController.java:511]: hmmm, i have been wondering, is this even necessary? all the SQL commands here are executed in `SqlScriptRunner`, which is not instrumented. so in theory they commands should be not recorded, regardless of this code, isn't it?

**Review by arcuri82 — CHANGES_REQUESTED** [review]

**Comment by ZhangMan1126** [comment]
checked it with rest-news using external driver. SQL for initialization could be ignored correctly.

**Pull request merged**